yarn-project/stdlib/src/messaging/inbox_leaf.ts

```diff
@@ -35,4 +35,15 @@ export class InboxLeaf {
     const end = start + BigInt(NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP);
     return [start, end];
   }
+
+  /** Returns the L2 block number for a given leaf index */
+  static l2BlockFromIndex(index: bigint): number {
+    return Number(index / BigInt(NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP)) + INITIAL_L2_BLOCK_NUM;
+  }
+
+  /** Returns whether a message with the given leaf index is synced */
+  static isMessageSynced(index: bigint, l2BlockNumber: number): boolean {
+    const messageL2Block = this.l2BlockFromIndex(index);
+    return messageL2Block <= l2BlockNumber;
```

Comment:
Think you can get away with
```ts
return messageL2Block <= l2BlockNumber + 1;
```
since the messages should be inserted at the start of the block, so can work if it is the next block. But this is stricter so should also work well, just slightly longer than needed.